lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py

```diff
@@ -0,0 +1,209 @@
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+
+
+class VmwareLockdownManager(PyVmomi):
+    def __init__(self, module):
+        super(VmwareLockdownManager, self).__init__(module)
+        if not self.is_vcenter():
+            self.module.fail_json(msg="Lockdown operations are performed from vCenter only. "
+                                      "hostname %s is an ESXi server. Please specify hostname "
+                                      "as vCenter server." % self.module.params['hostname'])
+        cluster_name = self.params.get('cluster_name', None)
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.hosts = []
+        if cluster_name:
+            cluster_obj = self.find_cluster_by_name(cluster_name=cluster_name)
+            if cluster_obj:
+                self.hosts = [host for host in cluster_obj.host]
+            else:
+                module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
+        elif esxi_host_name:
+            for host in esxi_host_name:
+                esxi_host_obj = self.find_hostsystem_by_name(host_name=host)
+                if esxi_host_obj:
+                    self.hosts.append(esxi_host_obj)
+                else:
+                    module.fail_json(changed=False, msg="ESXi '%s' not found" % host)
+
+    def ensure(self):
+        """
+        Function to manage internal state management
+        """
+        results = dict(changed=False, host_lockdown_state=dict())
+        change_list = []
+        for host in self.hosts:
+            changed = False
+            desired_state = self.params.get('state')
+            results['host_lockdown_state'][host.name] = dict(current_state='',
+                                                             desired_state=desired_state,
+                                                             previous_state=''
+                                                             )
+            try:
+                if host.config.adminDisabled:
+                    results['host_lockdown_state'][host.name]['previous_state'] = 'present'
+                    if desired_state == 'absent':
+                        host.ExitLockdownMode()
+                        results['host_lockdown_state'][host.name]['current_state'] = 'absent'
+                    else:
+                        results['host_lockdown_state'][host.name]['current_state'] = 'present'
+                elif not host.config.adminDisabled:
+                    results['host_lockdown_state'][host.name]['previous_state'] = 'absent'
+                    if desired_state == 'present':
+                        host.EnterLockdownMode()
+                        results['host_lockdown_state'][host.name]['current_state'] = 'present'
+                    else:
+                        results['host_lockdown_state'][host.name]['current_state'] = 'absent'
+                changed = True
```

Comment:
You did half of what I suggested :)
`changed = False` should be at the top, and `changed = True` should follow host.ExitLockdownMode and ost.EnterLockdownMode.